SG14_test/inplace_function_test.cpp

```diff
@@ -489,6 +489,90 @@ static void test_return_by_move()
     assert(InstrumentedCopyConstructor::moves == 1);
 }
 
+struct NoDefaultCtor
+{
+    int val;
+    explicit NoDefaultCtor(int v) : val{v} {}
+};
+
+int overloaded_function(stdext::inplace_function<int()>&& exec)
```

Comment:
Every function added in this PR should be `static`. Adding `namespace {` ... `}` around the definition of `struct NoDefaultCtor` is optional IMHO but still probably a good idea.